File: easy/118. Pascal's Triangle/118. Pascal's Triangle.ts
function generate(numRows: number): number[][] {
    const triangle = Array.from({ length: numRows }, (_, i) => new Array<number>(i + 1).fill(1))

    for (let i = 2; i < numRows; i++) {
        for (let j = 1; j < i; j++) {
            triangle[i][j] = triangle[i - 1][j - 1] + triangle[i - 1][j]
        }
    }

    return triangle
};

File: easy/118. Pascal's Triangle/118. Pascal's Triangle.py
from typing import List


class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        triangle = [[1] * i for i in range(1, numRows + 1)]

        for i in range(2, numRows):
            for j in range(1, i):
                triangle[i][j] = triangle[i - 1][j - 1] + triangle[i - 1][j]

        return triangle


solution = Solution()
print(solution.generate(numRows=5))
